### ISSUE TITLE
background-image with linear-gradient not shown in medium blogpost

### ISSUE THREAD

**Issue by CYBAI** [servo/servo#19810]
URL: https://slack.engineering/f56a5995e8f1

The first word, `webpack`, in the blog post also has this issue but there're more anchors with `text-decoration: underline` in the `Further reading` section; thus, I screenshot the section.

In correct way, we should see the underline below the anchors which have the CSS of `text-decoration`.

In Firefox Nightly:
<img width="1021" alt="default" src="https://user-images.githubusercontent.com/6782666/35107576-483d5fa6-fcac-11e7-9a63-bf86b0a37e6f.png">

In Servo (built with aee0d694cc):
<img width="1013" alt="default" src="https://user-images.githubusercontent.com/6782666/35107531-2bebd4ae-fcac-11e7-85d2-fbf2a1d6455d.png">

Tested in `macOS 10.12.6`

**Comment by CYBAI** [comment]
Here's the simplified test case and I'm going to change the title due to not issue of `text-decoration`.

```html
<html>
  <style>
    a {
      text-decoration: none;
      background-image: linear-gradient(to bottom,rgba(0,0,0,.68) 50%,rgba(0,0,0,0) 50%);
      background-repeat: repeat-x;
      background-size: 2px .1em;
      background-position: 0 1.07em;
  }
  </style>
  <a href="https://webpack.js.org/">webpack</a> is a brilliant tool for bundling frontend assets.
</html>
```

![image](https://user-images.githubusercontent.com/6782666/35109529-a2929b7e-fcb1-11e7-9add-f6e414a91457.png)

**Comment by emilio** [comment]
Thanks a lot for the test-case @CYBAI!

cc @pyfisch, this is the kind of stuff you've looked into before :)

**Comment by emilio** [comment]
(That being said, that's a gross way of drawing an underline :P)

**Comment by CYBAI** [comment]
(Haha, before inspecting it, I won't notice it's an underline in this way :P)

**Comment by jdm** [comment]
Yeah, this reminds me a lot of https://github.com/servo/servo/issues/19626.

**Comment by pyfisch** [comment]
> (That being said, that's a gross way of drawing an underline :P)

[They are kinda proud of it at Medium.](https://medium.design/crafting-link-underlines-on-medium-7c03a9274f9) :stuck_out_tongue: 

Unfortunately I can not reproduce it at the moment. Attached is the rendering as I am seeing it with an up-to-date master. (Medium itself does work too) I will try a nightly build if a new one is released.

![issue19810](https://user-images.githubusercontent.com/2781017/35126292-13011ae8-fcad-11e7-8556-fd28720f4364.png)

**Comment by pyfisch** [comment]
![medium](https://user-images.githubusercontent.com/2781017/35399558-ccaf8208-01f4-11e8-8264-2b1748b6512a.png)

I still see the underlines but the text is not correctly rendered. The article text has line-breaks at the wrong places and the heading is a different font.

jdm confirmed that the underlines are invisible on MacOS.

**Comment by jdm** [comment]
<img width="1136" alt="screen shot 2018-01-25 at 1 15 29 pm" src="https://user-images.githubusercontent.com/27658/35404685-d6595d2e-01d1-11e8-8ffa-94e28f4164a6.png">

**Comment by mrobinson** [comment]
This seems to be working now.
